tensorflow_io/core/kernels/azfs_kernels.cc

```diff
@@ -608,109 +692,189 @@ Status AzBlobFileSystem::DeleteDir(const std::string &dirname) {
   if (object.empty()) {
     blob_client.delete_container(container);
     if (errno != 0) {
-      return errors::Internal("Error deleting ", dirname, " (",
-                              errno_to_string(), ")");
+      std::string error_message =
+          absl::StrCat("Error deleting ", path, " (", errno_to_string(), ")");
+      TF_SetStatus(status, TF_INTERNAL, error_message.c_str());
+      return;
     }
   } else {
     // Delete all blobs under dirname prefix
     std::vector<std::string> children;
-    TF_RETURN_IF_ERROR(ListResources(dirname, "", blob_client, &children));
+    ListResources(path, "", blob_client, &children, status);
+    if (TF_GetCode(status) != TF_OK) {
+      return;
```

Comment:
Better to use TF_SetStatus here to explain that the deletion has not happened due to an error while listing resources.

Reply:
@kvignesh1420 It is less obvious though the `status` returned by `ListResource` set both the code and the error message. `TF_GetCode(status)` is to check if the status is bad. If status != OK then return immediately.